src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -2447,6 +2447,58 @@ int P
             return i;
         }
     }
+}");
+        }
+
+        [WorkItem(47999, "https://github.com/dotnet/roslyn/issues/47999")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public async Task TestPropertyIsReadOnlyAndSetterNeeded()
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    [|int i|];
+    public readonly int P => i;
+    public void SetP(int value) => i = value;
+}",
+@"class Class
+{
+    public int P { get; private set; }
+    public void SetP(int value) => P = value;
+}");
+        }
+
+        [WorkItem(47999, "https://github.com/dotnet/roslyn/issues/47999")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public async Task TestPropertyIsReadOnlyWithNoAccessModifierAndSetterNeeded()
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    [|int i|];
+    readonly int P => i;
+    public void SetP(int value) => i = value;
+}",
+@"class Class
+{
+    int P { get; set; }
+    public void SetP(int value) => P = value;
+}");
+        }
+
+        [WorkItem(47999, "https://github.com/dotnet/roslyn/issues/47999")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public async Task TestPropertyIsReadOnlyAndSetterUnneeded()
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    [|int i|];
+    public readonly int P => i;
+}",
+@"class Class
+{
+    public readonly int P { get; }
```

Comment:
Am I missing something? This isn't valid C#. Shouldn't all of these test cases be structs?

Reply:
Oops, my bad. I'll change that. Good catch! 😉